### ISSUE TITLE
I'm curious about the sequence for a graceful shutdown between the overall components of PINOT.

### ISSUE THREAD

**Issue by minwoo-jung** [apache/pinot#12234]
hello
We are deploying/operating pinot through helm chart.

# Problem

We restarted helm chart to upgrade pinot,
After restarting, some consuming segments of the realtime table were not consuming.
So I used the pause/resume api to solve the problem.

I am experiencing a similar problem sometimes when restarting pinot.
I would like to know how to prevent the problem from recurring.

### question 1
When we restart pinot, we restart the whole pinot with helm uninstall command.
Is there a startup or shutdown order for each PINOT component when restarting?
**In other words, I'm curious about the sequence for a graceful shutdown(or Rolling Restart) between the overall components of PINOT.**

For example, if you restart hbase, you usually need to restart it in the following order
- shutdown : hbase down -> hdfs down -> zookeeper down 
- start : zookeeper start -> hdfs start -> hbase start

### qeustion 2

Is there anything else I need to do on restart to avoid the above issue?
If so, I'd be curious to know.

**Comment by walterddr** [comment]
you can checkout the operational documents in https://docs.pinot.apache.org/operators/operating-pinot/setup-cluster

generally speaking you should ensure the ordering is controller/broker/server on start

**Comment by minwoo-jung** [comment]
@walterddr 
Thank you for your response.
When running a service that is composed of multiple components, we have found that the order of shutdown is very important.

**When shutting down, 
would it be correct to shut down in the order server -> broker -> controller as opposed to the order at startup?**

**Comment by walterddr** [comment]
AFAIK there's no shutdown ordering required. but i would recommend shutting down controller last b/c that's the metadata manager, if you shut it down earlier than servers. it is likely you will have to rewind your streaming ingestion to when the controller shutdown and restart from there.

**Comment by minwoo-jung** [comment]
@walterddr 
Thank you for your response.

Are you a core developer of pinot?
**I would like to ask a question to the core developers, in what cases consuming may not work properly when restarting.**

**Comment by walterddr** [comment]
for q&a's we generally encourage to ask in the slack channel https://github.com/apache/pinot?tab=readme-ov-file#join-the-community
